src/Generators/Microsoft.Gen.MetricsReports/Directory.Build.props

```diff
@@ -0,0 +1,9 @@
+<Project>
+  <!-- In order to get the right package versions for projects that shouldn't stabilize, we need to set this property before
```

Comment:
This is unfortunate, it defines the stage property in an inconsistent place and makes it harder to use correctly. Can we do something else like have these projects override the DotNetFinalVersionKind property?

Reply:
Using Directory.Build.props on library-specific projects to work around things like this is not that uncommon, in fact [dotnet/runtime use this all the time](https://github.com/search?q=repo%3Adotnet%2Fruntime+path%3Adirectory.build.props+path%3A%2F%5Esrc%5C%2Flibraries%5C%2F%2F&type=code). One more issue is that the reason why this property (DotnetFinalVersionKind) needs to be defined so early is because it needs to be defined before the import of `Version.BeforeCommonProps.props, which gets imported before the project, meaning that even if we want to override that property in the .csprojs it would still be too late for some other things that are derived from it.